src/object/mod.rs

```diff
@@ -306,6 +307,16 @@ mod traits {
         fn root<'a, C: Context<'a>>(&self, cx: &mut C) -> Root<Self> {
             Root::new(cx, self)
         }
+
+        fn call_method_with<'a, C, K>(&self, cx: &mut C, method: K) -> NeonResult<CallOptions<'a>>
+        where
+            C: Context<'a>,
+            K: PropertyKey,
+        {
+            let mut options = self.get::<JsFunction, _, _>(cx, method)?.call_with(cx);
+            options.this(JsValue::new_internal(self.to_raw()));
```

Comment:
Flagging this for follow-up. `JsValue::new_internal` should be marked `unsafe` since it allows arbitrarily extending a lifetime.

This usage of it is *safe* because it's bound by a new `Context` which will always be the shortest possible.